640/header

MarekBodingerBA opened the pull request:
Commit by commit.

TODOs:
 - mobile breadcrumbs

radoslavzeman commented on strapi/src/components/blocks/page-header-button.json:
```diff
@@ -0,0 +1,26 @@
+{
+  "collectionName": "components_blocks_page_header_buttons",
```
Please change `button` to `link`

radoslavzeman commented on strapi/src/api/page/content-types/page/schema.json:
```diff
@@ -132,6 +140,26 @@
+      },
+      "component": "blocks.page-header-button"
+    },
     "pageButtonContent": {
```
This old `pageButtonContent` acted as primary header button. It may need some migration or at least a table of page where it was used.

MarekBodingerBA replied:
Now it's merged, and `pageButtonContent` is used as a first button when available. I will remove them when migrated.

radoslavzeman commented on next/components/ui/PageHeaderV2/PageHeaderV2.tsx:
```diff
@@ -0,0 +1,97 @@
+            )}
+            {(title || subtitle) && (
+              <div className="gap-y-1 lg:gap-y-4 flex flex-col max-w-[800px]">
+                {title && <h1 className="text-h1">{title}</h1>}
```
Is there a page without `title`? There `must` be `<h1>` on the page, so I would make it required.

MarekBodingerBA replied:
It's just presentational and easier because of `null | undefined`.

radoslavzeman reviewed (commented):
Sorry if this wasn't written anywhere:
If there is one button, it should always be filled.
It there are two, first is filled, second is outlined.

Because of this requirement, we discussed that we can set up these buttons in Strapi as repeatable component with max 2 items. (Instead of having separate `primaryButton` and `secondaryButton`.

MarekBodingerBA commented on strapi/src/api/page/content-types/page/schema.json:
```diff
@@ -58,6 +58,14 @@
       },
       "type": "text"
     },
+    "subtitle": {
```
Done.

MarekBodingerBA commented on strapi/src/components/blocks/page-header-button.json:
```diff
@@ -0,0 +1,26 @@
+{
+  "collectionName": "components_blocks_page_header_buttons",
```
Used common link for the component.

MarekBodingerBA commented:
> Sorry if this wasn't written anywhere: If there is one button, it should always be filled. It there are two, first is filled, second is outlined.
> 
> Because of this requirement, we discussed that we can set up these buttons in Strapi as repeatable component with max 2 items. (Instead of having separate `primaryButton` and `secondaryButton`.

Added repeatable component and used common link, added blog post to common link.

radoslavzeman commented on next/components/ui/PageHeader/PageHeader.tsx:
```diff
@@ -70,14 +67,14 @@ const PageHeader = ({
               </div>
             )}
-            {(primaryButton || secondaryButton) && (
+            {buttons && buttons.length > 0 && (
```
`buttons?.length` is not good?
